lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -207,18 +207,23 @@ public Location parentDirectory()
      * Creates a new location by appending the given path element to the current path.
      * A slash will be added between the current path and the new path element if needed.
      *
-     * @throws IllegalArgumentException if the new path element is empty or starts with a slash
+     * @throws IllegalArgumentException if the new path element is empty
```

Comment:
I don't think we should be changing `Location.appendPath` API.
This can unwanted consequences. More conservative approach would be to keep disallowing leading `/` here and fix listing instead.